runtime/src/main/java/org/corfudb/protocols/wireprotocol/IMetadata.java

```diff
@@ -78,10 +79,30 @@ default Long getRank() {
      * @param rank The rank of this append.
      */
     default void setRank(long rank) {
+        if (rank>=0 && getRank()<0) {
+            throw new IllegalStateException("Cannot set rank when the overwrite is permitted");
```

Comment:
I am not sure this rank is being used for anything other than quorum replication; I think it was prepared in advance for that purpose. no?